ntbs-service/Helpers/EpisodesExtensionMethods.cs

```diff
@@ -43,13 +51,16 @@ public static List<TreatmentPeriod> GroupEpisodesIntoPeriods(this IEnumerable<Tr
                         periodNumber++;
                     }
 
+                    currentTreatmentPeriod.PeriodStartDate = treatmentEvent.EventDate;
+
                     treatmentPeriods.Add(currentTreatmentPeriod);
                 }
                 // Otherwise append this event to the existing period
                 else
                 {
                     currentTreatmentPeriod.TreatmentEvents.Add(treatmentEvent);
                 }
+                currentTreatmentPeriod.PeriodEndDate = treatmentEvent.EventDate;
```

Comment:
This will work, and I'm happy if you want to leave it this way.

However, I think a slightly better way to do this would be to move the start/end date logic into the `TreatmentPeriod` class, and have an override when creating the post-mortem period. That way the logic is isolated, and it reduces the cognitive load of reading this method.

Reply:
I've moved it into the `TreatmentPeriod` class now, by adding a `AddTreatmentEvent` function